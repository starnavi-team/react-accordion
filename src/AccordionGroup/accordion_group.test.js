import React from 'react';
import { shallow } from 'enzyme';
import {
  Accordion,
  AccordionGroup,
  AccordionGroupTitle,
  AccordionGroupBody,
} from '../index';
import { StyledAccordionGroup } from './index';

describe('AccordionGroup', () => {
  it('Makes possible to set custom className', () => {
    const testAccordionGroup = shallow(
      <Accordion>
        <AccordionGroup id="tested" className="testing">
          <AccordionGroupTitle />
          <AccordionGroupBody />
        </AccordionGroup>
      </Accordion>,
    );
    expect(testAccordionGroup.find('#tested')).toHaveClassName('testing');
  });

  it('Makes possible to set custom styles', () => {
    const testAccordionGroup = shallow(
      <Accordion>
        <AccordionGroup id="tested" style={{ position: 'absolute' }}>
          <AccordionGroupTitle />
          <AccordionGroupBody />
        </AccordionGroup>
      </Accordion>,
    );
    expect(testAccordionGroup.find('#tested')).toHaveStyle('position', 'absolute');
  });

  it('styledAccordionGroup receives custom className', () => {
    const styled = shallow(<StyledAccordionGroup className="tested" />);
    expect(styled.hasClass('tested')).toBe(true);
  });

  it('styledAccordionGroup receives custom styles', () => {
    const styled = shallow(<StyledAccordionGroup className="tested" style={{ color: 'green' }} />);
    expect(styled.find('.tested')).toHaveStyle('color', 'green');
  });

  it('should render correctly', () => {
    const component = shallow(<AccordionGroup />);
    expect(component).toMatchSnapshot();
  });
});
