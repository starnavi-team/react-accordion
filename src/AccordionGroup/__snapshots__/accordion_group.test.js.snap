// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionGroup should render correctly 1`] = `<styled.div />`;
